butterfree/clients/cassandra_client.py

```diff
@@ -53,12 +59,48 @@ def __init__(
         user: Optional[str] = None,
         password: Optional[str] = None,
     ) -> None:
-        self.host = host
+        self.host = self._validate_and_format_cassandra_host(host)
+        logger.info(f"The host setted is {self.host}")
         self.keyspace = keyspace
         self.user = user
         self.password = password
         self._session: Optional[Session] = None
 
+    def _validate_and_format_cassandra_host(self, host: Union[List, str]):
+        """
+        Validate and format the provided Cassandra host input.
+
+        This method checks if the input `host` is either a string, a list of strings, or
+        a list containing a single string with comma-separated values. It splits the string
+        by commas and trims whitespace, returning a list of hosts. If the input is already
+        a list of strings, it returns that list. If the input is empty or invalid, a
+        ValueError is raised.
+
+        Args:
+            host (str | list): The Cassandra host input, which can be a comma-separated
+                            string or a list of string endpoints.
+
+        Returns:
+            list: A list of formatted Cassandra host strings.
+
+        Raises:
+            ValueError: If the input is an empty string or if it is not a string
```

Comment:
```suggestion
            ValueError: If the input is an empty list or if it is not a string
```

Reply:
Actually it's both. Empty string or list is the problem

Reply:
Updated